### ISSUE TITLE
salt.pillar.netbox: interfaces?

### ISSUE THREAD

**Issue by misch42** [saltstack/salt#51490]
### Description of Issue/Question

I set up the salt.pillar.netbox external pillar. eveything works fine. With `pillar.items` I get the data and with `'netbox:...'` I can filter for keys.

But: I do not see interface information in the pillar data.

Is is possible to retrieve the interface and VLAN data from netbox?

Michael

**Comment by Ch3LL** [comment]
ping @ksofa2 looks like you added this here: https://github.com/saltstack/salt/pull/46559

can you take a look here?

**Comment by ksofa2** [comment]
cc @ ngrundler

**Comment by ksofa2** [comment]
@misch42 It's just that interfaces weren't in the scope for our initial implementation of this pillar. We probably don't have time to add this anytime soon, but would be happy to lend a hand for review, etc if you or someone else puts together a PR

**Comment by misch42** [comment]
I thunk 2019.2.0 solves the request with the new function netbox.openconfig_interfaces

**Comment by eliezerlp** [comment]
> I think 2019.2.0 solves the request with the new function netbox.openconfig_interfaces

@misch42, seems that is only an execution module, not a pillar module... I think this issue aimed to get the data via Pillar.

**Comment by misch42** [comment]
A pillar that delivers openconfig would really by good.

**Comment by maznu** [comment]
@misch42: it's not openconfig, but the PR above does add interfaces and/or IP addresses to the netbox pillar. :-)

**Comment by eliezerlp** [comment]
Is PR https://github.com/saltstack/salt/pull/52704 where this is to be implemented? Should efforts be focused there?

**Comment by Ch3LL** [comment]
looks like that indeed is the PR that would resolve this issue. looks like its waiting on tests, and will also need to be rebased to master.

ping @misch42 anything i can do to help you get it over the line?

**Comment by Ch3LL** [comment]
sorry wrong ping, i mean @maznu

**Comment by sagetherage** [comment]
@Ch3LL putting this back into triage to review what is next, let me know if I can help

**Comment by Ch3LL** [comment]
Looks like its still a feature request that needs to be implemented since the PR was closed.

**Comment by misch42** [comment]
As far as I know, this still is not implemented. So this is a feature request.

**Comment by sagetherage** [comment]
I would love to see another PR opened against the `master` branch and has tests. I am putting community label on it for now to see if we can get community help for Aluminium release cycle.
